docs/dev-guide/api-ref.md

```diff
@@ -153,9 +153,163 @@ where `:clientId` is the `_id` property of the client to delete.
 
 The response code will be `200` if successful.
 
+### Roles resource
+
+Allows for the management of client access control to channels.
+
+It should be noted that there is no actual roles collection in the database. The API is a facade on top of the `allow` and `roles` fields from Channels and Clients respectively. Roles can therefore also be altered by changing values for those fields directly.
+
+#### Fetch all roles
+
+`GET /roles`
+
+The response status will be `200` if successful and the response body will contain an array of role objects, each consisting of a `name`, an array of `channels` and an array of `clients`, e.g.:
+```json
+[
+  {
+    "name": "Role1",
+    "channels": [
+      {
+        "_id": "56d56f34131d779a3f220d6d",
+        "name": "channel1"
+      },
+      {
+        "_id": "56dfff5ef51fbdc660fe6722",
+        "name": "channel2"
+      }
+    ],
+    "clients": [
+      {
+        "_id": "56d43e584582beae226d8226",
+        "clientID": "jembi"
+      }
+    ]
+  },
+  {
+    "name": "Role2",
+    "channels": [
+      {
+        "_id": "56d43e424582beae226d8224",
+        "name": "Channel3"
+      }
+    ],
+    "clients": [
+      {
+        "_id": "56d43e584582beae226d8226",
+        "clientID": "jembi"
+      }
+    ]
+  },
+  {
+    "name": "internal",
+    "channels": [
+      {
+        "_id": "56e116d9beabfb406e0e7c91",
+        "name": "Daily Task"
+      }
+    ],
+    "clients": []
+  }
+]
+```
+
+#### Fetch a specific role by role name
+
+`GET /roles/:name`
+
+The response status code will be `200` if successful and the response body will contain a role object in the same format as the role elements in the *Fetch all roles* operation response above. E.g.
+```js
+{
+  "name": "Role1",
+  "channels": [
+    {
+      "_id": "56d56f34131d779a3f220d6d",
+      "name": "channel1"
+    },
+    {
+      "_id": "56dfff5ef51fbdc660fe6722",
+      "name": "channel2"
+    }
+  ],
+  "clients": [
+    {
+      "_id": "56d43e584582beae226d8226",
+      "clientID": "jembi"
+    }
+  ]
+}
+```
+
+#### Add a new role
+
+`POST /roles`
+
+with a json body containing the role name and channels and clients to apply to. Clients do not need to be specified, but at least one channel has to. Channels and clients can be specified either by their `_id` or `name` for a channel and `clientID` for a client.
```

Comment:
If there are clients present does a channel still need to be added? If so, then I'm not sure if this is the best rule. Often users would want to create user and roles before assigning them to channels. At least that is how I would do it.